osu.Game/Overlays/Direct/PlayButton.cs

```diff
@@ -78,19 +79,50 @@ public PlayButton(BeatmapSetInfo setInfo = null)
                 loadingAnimation = new LoadingAnimation(),
             });
 
-            Playing.ValueChanged += updatePreviewTrack;
+            Playing.ValueChanged += playingStateChanged;
         }
 
+        private PreviewTrackManager previewTrackManager;
+
         [BackgroundDependencyLoader]
-        private void load(OsuColour colour, AudioManager audio)
+        private void load(OsuColour colour, PreviewTrackManager previewTrackManager)
         {
+            this.previewTrackManager = previewTrackManager;
+
             hoverColour = colour.Yellow;
-            this.audio = audio;
         }
 
         protected override bool OnClick(InputState state)
         {
-            Playing.Value = !Playing.Value;
+            if (!Playing.Value)
+            {
+                if (Preview != null)
+                {
+                    Preview.Start();
+                    return true;
+                }
+
+                loading = true;
+
+                var loadingPreview = previewTrackManager.Get(beatmapSet);
+                loadingPreview.Started += () => Playing.Value = true;
+                loadingPreview.Stopped += () => Playing.Value = false;
+
+                LoadComponentAsync(Preview = loadingPreview, t =>
+                {
+                    if (Preview != loadingPreview)
+                        return;
+
+                    AddInternal(t);
```

Comment:
either use `loadingPreview` here (and make `t` a `_`) or use `t` above and don't use `loadingPreview` anywhere.

Reply:
should probably be loadingPreview, sure